src/CommunityToolkit.Maui.SourceGenerators/Extensions/NamespaceSymbolExtensions.cs

```diff
@@ -26,4 +26,129 @@ public static IEnumerable<INamedTypeSymbol> GetNamedTypeSymbols(this INamespaceS
 			}
 		}
 	}
-}
\ No newline at end of file
+	
+	public static string GetFullTypeString(this INamedTypeSymbol type) => 
+		type.Name + 
+		(type.NullableAnnotation == NullableAnnotation.Annotated ? "?" : string.Empty) +
+		type.TypeArguments.GetGenericTypeArgumentsString();
+
+	public static string GetGenericTypeArgumentsString(this IEnumerable<ITypeSymbol> typeArguments)
+	{
+		string result = string.Empty;
```

Comment:
Can you please use a StringBuilder instead?

Reply:
yes, this makes total sense